sunpy/net/jsoc/tests/test_jsoc.py

```diff
@@ -318,3 +330,17 @@ def test_results_filenames():
     assert len(files) == len(responses)
     for hmiurl in files:
         assert os.path.isfile(hmiurl)
+
+
+@pytest.mark.remote_data
+def test_results_filenames_as_is():
+    responses = client.search(
+        a.Time('2014/1/1T1:00:36', '2014/1/1T01:01:38'),
+        a.jsoc.Series('hmi.M_45s'), a.jsoc.Notify('jsoc@cadair.com'),
+        a.jsoc.Protocol('as-is'))
+    path = tempfile.mkdtemp()
```

Comment:
rather than using `tempfile` like this it would be better to use the [`tmp_path`](https://docs.pytest.org/en/latest/tmpdir.html#the-tmp-path-fixture) pytest fixture.